pkg/container/types/embedding.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"math"
+	"strconv"
+	"strings"
+	"unsafe"
+)
+
+const (
+	// MaxEmbeddingDimension Based on here: https://github.com/pgvector/pgvector/blob/b56971febeec389a011de7bb40b3349e24757aff/src/vector.h#L10
+	MaxEmbeddingDimension = 16000
+)
+
+func BytesToEmbedding(input []byte) (res []float32) {
+	if len(input)%4 != 0 {
+		panic(moerr.NewInternalErrorNoCtx("the byte slice length must be a multiple of 4"))
+	}
+
+	dimension := len(input) / 4
```

Comment:
Considering the different types supported, it could probably be written slightly more generically here, 4 can be a argument